gravitee-apim-definition/gravitee-apim-definition-model/src/main/java/io/gravitee/definition/model/federation/FederatedAgent.java

```diff
@@ -0,0 +1,125 @@
+ */
+package io.gravitee.definition.model.federation;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.gravitee.definition.model.DefinitionVersion;
+import jakarta.annotation.Nullable;
+import jakarta.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+
+// https://google.github.io/A2A/specification/#5-agent-discovery-the-agent-card
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder(toBuilder = true)
+public class FederatedAgent implements Serializable {
+
+    public static final String STREAMING = "streaming";
+    public static final String PUSH_NOTIFICATIONS = "pushNotifications";
+    public static final String STATE_TRANSITION_HISTORY = "stateTransitionHistory";
+
+    private String name;
+    private String description;
+    private String url;
+    private String version;
+
+    @Nullable
+    private String documentationUrl;
+
+    @Nullable
+    private Provider provider;
+
+    private Map<String, Boolean> capabilities;
+    private Collection<Skill> skills;
+
+    //@Nullable
+    //private Map<String, SecurityScheme> securitySchemes;
+
+    //@Nullable
+    //private Map<String, Collection<String>> security;
+
+    @JsonProperty(required = true)
+    @NotNull
+    @Builder.Default
+    private DefinitionVersion definitionVersion = DefinitionVersion.FEDERATED_AGENT;
+
+    public record Provider(String organization, String url) {}
+
+    public record Skill(
+        String id,
+        String name,
+        Collection<String> tags,
+        @Nullable Collection<String> examples,
+        @Nullable Collection<String> inputModes,
+        @Nullable Collection<String> outputModes
+    ) {}
+
+    // https://swagger.io/specification/#security-scheme-object
+    public sealed interface SecurityScheme {
+        Type type();
+
+        String description();
+
+        record APIKey(String description, String name, String in) implements SecurityScheme {
+            public Type type() {
+                return Type.API_KEY;
+            }
+        }
+
+        record HTTPAuth(String description, String scheme, String bearerFormat) implements SecurityScheme {
+            public Type type() {
+                return Type.HTTP;
+            }
+        }
+
+        record OAuth2(String description) implements SecurityScheme {
+            public Type type() {
+                return Type.OAUTH;
+            }
+        }
+
+        record OpenIdConnect(String description, String openIdConnectUrl) implements SecurityScheme {
+            public Type type() {
+                return Type.OPEN_ID;
+            }
+        }
+
+        record MutualTLS(String description) implements SecurityScheme {
+            public Type type() {
+                return Type.MTLS;
+            }
+        }
```

Comment:
MTLS seems not supported by the protocol https://github.com/google/A2A/blob/main/types/src/types.ts#L360